apps/condo/bin/migrate-user-import-id-data-to-user-external-identity.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Migrate User.importId/User.importRemoteSystem fields data to separate entity: UserExternalIdentity
+ *
+ * Usage:
+ *      yarn workspace @app/condo node bin/migrate-user-import-id-data-to-user-external-identity
+ */
+
+const path = require('path')
+const dayjs = require('dayjs')
+
+const { GraphQLApp } = require('@keystonejs/app-graphql')
+const { getSchemaCtx } = require('@open-condo/keystone/schema')
+
+const { User, UserExternalIdentity } = require('@condo/domains/user/utils/serverSchema')
+const { IDP_TYPES } = require('@condo/domains/user/constants/common')
+
+
+// init utils & common vars
+const dv = 1
+const sender = { dv, fingerprint: 'migrate-user-importId-to-externalIdentity' }
+
+const log = (args) => {
+    console.log(`${new Date().toISOString()}: ${JSON.stringify(args)}`)
+}
+
+const readImportedUsersPage = async ({ context, offset, pageSize }) => {
+    // retrieve users page where user.importId is filled up
+    return await User.getAll(context, { importId_not: null, importRemoteSystem_not: null }, {
+        sortBy: 'id_ASC',
+        first: pageSize,
+        skip: offset,
+    })
+}
+
+const migrateUserImportIdDataToUserExternalIdentity = async (batchSize = 100) => {
+    // initialize context
+    const { keystone: context } = await getSchemaCtx('User')
+
+    // let's save process state to provide more readable logs
+    let usersBatch = []
+    const state = {
+        pageSize: batchSize,
+        offset: 0,
+        hasMore: true,
+        processedPages: 0,
+        migratedUsers: 0,
+        alreadyMigratedUsers: 0,
+        usersWithNotExpectedRemoteSystemCode: 0,
+        notExpectedRemoteSystemCodes: [],
```

Comment:
You can use Set for duplicate-proof storage

Reply:
this is the state of processing process. I'm going to re-calculate state based on current state values. It not supposed to have duplicates at all